### DISCUSSION TITLE
Support for filling forms

### DISCUSSION THREAD

**Discussion by filamento** [sumatrapdfreader/sumatrapdf#2386] [Ideas]
Dear Sir,

Could you add support for editing and filling forms like in this sample PDF ??

[SEPA_particulares_es.pdf](https://github.com/sumatrapdfreader/sumatrapdf/files/8161329/SEPA_particulares_es.pdf)

It would be very nice, we could type them in the computer before printing and would avoid having to fill them laboriously by hand!

**Comment by GitHubRulesOK** [comment]
Hmm ..
there are many ways a Form can be defined and SumatraPDF is based on MuPDF abilities, which itself does include form field entry,
![image](https://user-images.githubusercontent.com/12754913/156167797-d142a478-554c-42ca-a884-43a736515b2a.png) 
![image](https://user-images.githubusercontent.com/12754913/156167854-1aefefda-f617-437f-a9a8-2abbd81dbdf6.png)

Long term the aim is for SumatraPDF to work with at least forms similar to that one , however , currently it is possible for SumatraPDF to overlay text on a form, BUT that form does not allow text overlay in the fields !

![image](https://user-images.githubusercontent.com/12754913/156168349-47b58371-af39-4501-b60c-482130b7a1a7.png)

**Reply by filamento** [reply]
Yes, they are a bit weird. But Microsoft's Edge browser (the new one, based on Chromium) seems to be able to fill them without problems.

I think it could be a very nice addition for Sumatra. Maybe you could add this to your roadmap for a future version?

**Comment by GitHubRulesOK** [comment]
I would be using use Edge for the filling, it is possible for SumatraPDF to open that page in a companion copy of Edge for inking etc.
and SumatraPDF developer has indicated a wish to at least match the MuPDF form ability, 
but there are many higher priorities than that one.

**Comment by kjk** [comment]
We already have an issue for tracking this: https://github.com/sumatrapdfreader/sumatrapdf/issues/1249
